src/main/java/com/github/mkopylec/projectmanager/application/ProjectService.java

```diff
@@ -81,4 +88,19 @@ public void startProject(String projectIdentifier) {
         project.start();
         projectRepository.save(project);
     }
+
+    public void endProject(String projectIdentifier, ProjectEndingCondition endingCondition) {
+        Project project = projectRepository.findByIdentifier(projectIdentifier);
+        when(project == null)
+                .thenMissingEntity(NONEXISTENT_PROJECT, "Error ending '" + projectIdentifier + "' project");
+        FeatureChecker featureChecker = resolveFeatureChecker(endingCondition.isOnlyNecessaryFeatureDone());
+        EndedProject endedProject = project.end(featureChecker);
+        if (project.hasAssignedTeam()) {
+            Team assignedTeam = teamRepository.findByName(project.getAssignedTeam());
+            assignedTeam.removeCurrentlyImplementedProject();
+            teamRepository.save(assignedTeam);
```

Comment:
It should be second if repository project fails.

teamRepository.save(assignedTeam);